index.bs

```diff
@@ -11852,34 +12105,131 @@ must be {{%IteratorPrototype%}}.
-                1.  Call <a abstract-op>CreateDataProperty</a>(|array|, "<code>1</code>", |value|).
-                1.  |result| is |array|.
-        </dl>
-    1.  Return <a abstract-op>CreateIterResultObject</a>(|result|, <emu-val>false</emu-val>).
+    1.  Return the [=iterator result=] for |pair| and |kind|.
 </div>
 
 The [=class string=] of an [=iterator prototype object=] for a given [=interface=]
 is the result of concatenating the [=identifier=] of the [=interface=]
 and the string "<code> Iterator</code>".
 
 
+<h5 id="es-default-asynchronous-iterator-object">Default asynchronous iterator objects</h5>
+
+A <dfn id="dfn-default-asynchronous-iterator-object" export>default asynchronous iterator object</dfn>
+for a given [=interface=], target and iteration kind is an object whose \[[Prototype]]
+[=internal slot=] is the [=asynchronous iterator prototype object=] for the [=interface=].
+
+A [=default asynchronous iterator object=] has internal values:
+
+*   its <em>target</em>, which is an object whose values are to be iterated,
+*   its <em>kind</em>, which is the iteration kind,
+*   its <em>ongoing promise</em>, which is a {{Promise}},
+*   its <em>state</em>, which is an opaque value used to store the position of the iterator by the
+    algorithm to [=get the next iteration result=], or null.
+
+When a [=default asynchronous iterator object=] is first created, its state is not given.
+
+Note: [=Default asynchronous iterator objects=] do not have [=class strings=]; when
+<code class="idl">Object.prototype.toString()</code> is called on a
+[=default asynchronous iterator object=] of a given [=interface=], the [=class string=] of the
+[=asynchronous iterator prototype object=] of that [=interface=] is used.
+
+
+<h5 id="es-asynchronous-iterator-prototype-object">Asynchronous iterator prototype object</h5>
+
+The
+<dfn id="dfn-asynchronous-iterator-prototype-object" export>asynchronous iterator prototype object</dfn>
+for a given [=interface=] is an object that exists for every interface that has an
+[=asynchronously iterable declaration=].
+It serves as the prototype for [=default asynchronous iterator objects=] for the interface.
+
+The \[[Prototype]] [=internal slot=] of an [=asynchronous iterator prototype object=] must be
+{{%AsyncIteratorPrototype%}}.
+
+<div algorithm="to invoke the next property of asynchronous iterators">
+
+    An [=asynchronous iterator prototype object=] must have a <code class="idl">next</code> data
+    property with attributes
+    { \[[Writable]]: <emu-val>true</emu-val>, \[[Enumerable]]: <emu-val>true</emu-val>, \[[Configurable]]: <emu-val>true</emu-val> }
+    and whose value is a [=built-in function object=] that behaves as follows:
+
+    1.  Let |interface| be the [=interface=] for which the
+        [=asynchronous iterator prototype object=] exists.
+
+    1.  Let |thisValidationPromiseCapability| be [=!=] [$NewPromiseCapability$]({{%Promise%}}).
```

Comment:
If we had a PromiseReject abstract op, you could just create and return a new promise inline each time. I guess we'd wait for #490 for that.